### PR TITLE
CI Updates

### PR DESCRIPTION
**Pull request by esc**
As title

### CODE REVIEW

**Comment by esc** [comment]
This updates the makefile such that the targets can be used on the command line when developing and during the GHA when doing continuous integration (CI).

Also, we fold the pre-commit GHA into the main testing, rather than having this a seprarate task.

File: makefile [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
-	pre-commit run --all-files
+	pre-commit run --verbose --all-files
 docs:
 	cd docs && make html
 conda-env:
-	conda create -n numba-rvsdg python=3.12 python-graphviz pyyaml pytest sphinx sphinx_rtd_theme
+	conda create -n numba-rvsdg
+conda-install:
+	conda install python=3.12 python-graphviz pyyaml pytest sphinx sphinx_rtd_theme coverage pre-commit
+	pip install -U virtualenv
```
Comment (skupr-anaconda) [inline makefile:17]: @esc I wonder if `conda install virtualenv` will work too...

Comment (esc) [reply makefile:17]: I tried that, but when I use `pip` I get `virtualenv-20.26.1` anaconda.org only has `20.17.1` which seems to old.

**Review by skupr-anaconda — APPROVED** [review]
LGTM

**Comment by esc** [comment]
> LGTM

Thank you for the review, I added two more snippets to update some dependencies and also making a note about the need for `pip install virtualenv`.

**Comment by esc** [comment]
@skupr-anaconda ❤️

**Pull request merged**